pkg/cloudprovider/providers/gce/metrics.go

```diff
@@ -67,6 +68,12 @@ func (mc *metricContext) Observe(err error) error {
 }
 
 func newGenericMetricContext(prefix, request, region, zone, version string) *metricContext {
+	if len(strings.TrimSpace(zone)) == 0 {
```

Comment:
Why do we have to sanitize the strings here? I'm surprised that we are getting somewhat invalid input within the code we control.